### PR TITLE
Add GCE Router task

### PR DESCRIPTION
**Pull request by kenji-cloudnatix**
This commit picks up the change from the previous attempt
(https://github.com/kubernetes/kops/pull/6828).

- Add Router to GCE tasks
- Add the `HasExternalIP` field to `InstanceTemplate`
- Create a `Router task and set `HasExternalIP` to `false` when a private topology is specified.

https://github.com/kubernetes/kops/issues/6827

### CODE REVIEW

**Comment by rifelpet** [comment]
this looks good to me! Could we add an integration test for this? it may require some changes in the cloudmock directory, I can provide assistance for that if needed.

**Comment by kenji-cloudnatix** [comment]
Thanks! Let me take a look. Will get back if I have any question!

**Comment by kenji-cloudnatix** [comment]
Hi @rifelpet , took a while but added an integration test with [a separate PR](https://github.com/kubernetes/kops/pull/11291). Does the direction look good to you?

**Comment by hakman** [comment]
@kenji-cloudnatix I think for this should be enough to add a TF output like:
https://github.com/kubernetes/kops/tree/master/tests/integration/update_cluster/minimal_gce

**Comment by kenji-cloudnatix** [comment]
Sounds good! Just updated. @hakman ,  @rifelpet , could you take a look again?

File: upup/pkg/fi/cloudup/gcetasks/instancetemplate.go [hunk 1]
```diff
@@ -133,6 +140,17 @@ func (e *InstanceTemplate) Find(c *fi.Context) (*InstanceTemplate, error) {
...
 			}
+
+			acs := ni.AccessConfigs
+			if len(acs) > 0 {
+				if len(acs) != 1 {
+					return nil, fmt.Errorf("unexpected number of access configs: %d", len(acs))
+				}
+				if acs[0].Type != accessConfigOneToOneNAT {
+					return nil, fmt.Errorf("unexpected access type: %s", acs[0].Type)
+				}
```
Comment (hakman) [inline upup/pkg/fi/cloudup/gcetasks/instancetemplate.go:151]: Maybe mention InstanceTemplate name in error messages.

**Review by hakman — CHANGES_REQUESTED** [review]
Thanks @kenji-cloudnatix. Just a small nit, other than that looks good.
/lgtm

File: upup/pkg/fi/cloudup/gcetasks/instancetemplate.go (hunk 1 above)
Comment (kenji-cloudnatix) [reply upup/pkg/fi/cloudup/gcetasks/instancetemplate.go:151]: done!

**Review by kenji-cloudnatix — COMMENTED** [review]
Thanks for the review!

**Review by hakman — APPROVED** [review]

**Pull request merged**